### PR TITLE
[WWST-2406] Updated Aeon Multisensor DTH to support Zooz 4-in-1 sensor

### PR DESCRIPTION
**Pull request by PKacprowiczS**
@greens @MMateusiakS @KKlimczukS @DCzarneckaS @MGoralczykS

### CODE REVIEW

**Comment by greens** [comment]
Looks like this device supports tamper as well, so multisensor 6 is probably a better fit.

File: devicetypes/smartthings/aeon-multisensor.src/aeon-multisensor.groovy [hunk 1]
```diff
@@ -236,4 +263,20 @@ private setConfigured(configure) {
 
 private isConfigured() {
 	getDataValue("configured") == "true"
+}
+
+private sendCheckInterval() {
+	def checkInterval
+	if(zwaveInfo.mfr == "027A") checkInterval = 8 * 60 * 60 + 2 * 60
```
Comment (kianooshkarami) [inline devicetypes/smartthings/aeon-multisensor.src/aeon-multisensor.groovy]: The formula going forward for Z-Wave sleepy devices should be `(2 * Wake up Interval ) + ( 10 minutes) `

Comment (kianooshkarami) [reply devicetypes/smartthings/aeon-multisensor.src/aeon-multisensor.groovy]: I would suggest to set the `def wakeupInterval` and then before sending the check interval do ` checkInterval = 2 * checkInterval + (10 * 60)`

File: devicetypes/smartthings/aeon-multisensor.src/aeon-multisensor.groovy [hunk 2]
```diff
@@ -110,17 +122,32 @@ def parse(String description)
...
+		createEvent(descriptionText: cmd.toString())
+	}
+}
+
 def zwaveEvent(physicalgraph.zwave.commands.wakeupv1.WakeUpNotification cmd)
 {
-	[descriptionText: "${device.displayName} woke up", isStateChange: false]
+	def cmds = []
+	def result = createEvent(descriptionText: "$device.displayName woke up", isStateChange: false)
+	if (!state.lastbatt || (now() - state.lastbatt) >= 10 * 60 * 60 * 1000) {
```
Comment (kianooshkarami) [inline devicetypes/smartthings/aeon-multisensor.src/aeon-multisensor.groovy]: Going forward and Per Z-Wave Plus V2 rules, we need to ask all the sleepy nodes for their battery level when they wake up. So I would say, let's just keep it simple, and just ask for Battery level every time the device wakes up

**Comment by PKacprowiczS** [comment]
> Looks like this device supports tamper as well, so multisensor 6 is probably a better fit.

@greens but multisensor 6 has also UV index and power source, which this device lacks.

File: devicetypes/smartthings/aeon-multisensor.src/aeon-multisensor.groovy (hunk 1 above)
Comment (PKacprowiczS) [reply devicetypes/smartthings/aeon-multisensor.src/aeon-multisensor.groovy]: Wasn't 4h a default wakeUpInterval time value for sleepy devices, or I have missed something in the meantime?

Comment (kianooshkarami) [reply devicetypes/smartthings/aeon-multisensor.src/aeon-multisensor.groovy]: It does not appear that we are configuring the Wake up interval to be something other than 4 hours, I thought we were, sorry about the confusion. Having said that, we should have the same checkInterval for all devices using this DTH. 

We should use this formula going forward for all Z-Wave sleepy devices `checkInterval = 2 * checkInterval + (10 * 60)`.

**Comment by greens** [comment]
I think I meant gen5. Most of the modernization you did in this PR (battery/wakeup/security) is already in gen 5.

**Comment by PKacprowiczS** [comment]
In that case you're right, but gen 5 doesn't support tamper either, so now I'm not sure if I should add this device to existing DTH or create brand new with support for all features.

**Comment by greens** [comment]
You could add tamper and the fingerprint to the gen5 DTH and then generate and assign a device-specific metadata that includes tamper.

That way the Aeon device would not display the tamper tile in OneApp while the Zooz device would.

**Review by greens — COMMENTED** [review]
probably just revert all the changes on `aeon-multisensor`

**Comment by PKacprowiczS** [comment]
@greens reverted and squashed

File: devicetypes/smartthings/aeon-multisensor-gen5.src/aeon-multisensor-gen5.groovy [hunk 3]
```diff
@@ -21,9 +21,11 @@ metadata {
 		capability "Sensor"
 		capability "Battery"
 		capability "Health Check"
+		//capability "Tamper Alert" waiting for LE fix for Zooz 4-in-1
```
Comment (greens) [inline devicetypes/smartthings/aeon-multisensor-gen5.src/aeon-multisensor-gen5.groovy]: Sorry I didn't realize. This isn't the right approach. 

We should just give the Zooz fingerprint-specific metadata pointing to "generic-motion-6", which includes everything except Tamper

**Review by greens — APPROVED** [review]

**Comment by dkirker** [comment]
The function "clearTamper" is not defined...

**Comment by greens** [comment]
@dkirker that was fixed in the later PR: https://github.com/SmartThingsCommunity/SmartThingsPublic/pull/11083

**Comment by dkirker** [comment]
@greens I see. *that* one is still missing `clearTamper`... That also make more sense to the exception that Steven was seeing.

**Pull request merged**